src/app/drive/components/ShareDialog/ShareDialog.tsx

```diff
@@ -259,7 +268,11 @@ const ShareDialog = (props: ShareDialogProps): JSX.Element => {
     }
 
     if (itemToShare?.item.uuid) {
-      await shareService.getPublicShareLink(itemToShare?.item.uuid, itemToShare.item.isFolder ? 'folder' : 'file');
+      await shareService.getPublicShareLink(
+        itemToShare?.item.uuid,
+        itemToShare.item.isFolder ? 'folder' : 'file',
+        (itemToShare?.item as unknown as AdvancedSharedItem)?.encryptionKey,
```

Comment:
Hey This does not come from the SDk. I think is more like a type problem as typescript does not know if this is driveItem or advanced Share Item. Added a typeguard function as TS recommends and a small change in the type to prevent this. 

I might be wrong tho, sorry if so.